inspirehep/modules/workflows/utils.py

```diff
@@ -141,3 +145,67 @@ def _try_to_log(logfn, *args, **kwargs):
         return res
 
     return _decorator
+
+
+@with_debug_logging
+def get_pdf_in_workflow(obj):
+    """Return the fullpath to the PDF attached to a workflow object
+
+    Return the PDF attached to a workflow object. If no PDF is already attached
+    and the workflow object is related to an arXiv record, fetches on the fly
+    the PDF from arXiv and attaches it.
+    """
+    for filename in obj.files.keys:
+        if filename.endswith('.pdf'):
```

Comment:
This is not how you detect if a file is a PDF. This is a bad idea for at least two reasons:

1. We already experienced issues where a file has the wrong extension, causing a later exception.
2. This also represents a security vulnerability, because it trusts user input about the format of a file. Basically this code is like the clueless user that clicks on `Very_Important.exe.doc`.